src/raftstore/store/worker/apply.rs

```diff
@@ -1163,18 +1166,13 @@ impl ApplyDelegate {
                 )
             });
 
-        // Use delete_range to mark all the contents in this range is deleted.
-        ctx.wb
-            .delete_range_cf(handle, &start_key, &end_key)
-            .unwrap_or_else(|e| {
-                panic!(
-                    "{} failed to delete range [{}, {}): {:?}",
-                    self.tag,
-                    escape(&start_key),
-                    escape(&end_key),
-                    e
-                )
-            });
+        // Delete all remaining keys.
+        try!(util::delete_all_in_range_cf(
```

Comment:
Better panic here.

Reply:
Is it safe to write the WriteBatch early here? /cc @javaforfun

Reply:
should write apply_state together

Reply:
seems no harm if node restart,  delete_range has been run, but apply_index not changed. 
since this node can't be leader after restart,  and if it become leader, the old apply_index (for delete_range)  already runned.